Task1/Project/Shapes/Models/Classes/ShapeBase.cs

```diff
@@ -8,8 +11,25 @@ namespace Shapes.Models.Classes
     /// </summary>
     public abstract class ShapeBase : IShape, IFileManager
     {       
+        // FIELDS
+        static IdentifierShape idOfTheShape;
+        static Dictionary<string, Deserialization> dictionaryOfId;    

```

Comment:
it is not **dictionaryOfId**, it is a **factories**, something that by key can create object

Reply:
Rewrote.